keps/sig-network/2091-admin-network-policy/README.md

```diff
@@ -402,182 +393,262 @@ The following new `AdminNetworkPolicy` API will be added to the `policy.networki
+  // SameLabels is used to select a set of Namespaces that share the same label(s).  
+  // To be selected a Namespace must have all of the labels defined in SameLabels, 
+  // If Samelabels is Empty then nothing is selected
+  SameLabels          []string
+  // NotSameLabels is used to select a set of Namespaces that do not have a set 
+  // of label(s). To be selected a Namespace must have none of the labels defined 
+  // in NotSameLabels. If NotSameLabels is empty then nothing is selected
+  NotSameLabels       []string
 }
 
-// NamespaceSetAndPod defines a way to select Namespaces and Pods in a cluster 
-type NamespaceSetAndPod struct {
-  Scope               NamespaceMatchType
-  // Labels should be set only when scope is "SameLabels" or "NotSameLabels.
-  // +optional
-  Labels              []string
-  // NamespaceSelector should be set only when scope is "Selector". 
-  NamespaceSelector   *metav1.LabelSelector
-  // PodSelector should be set only when scope is "Selector", "Self", or "NotSelf".
-  // This allows the user to select matching pods in the same namespace
-  PodSelector         *metav1.LabelSelector 
+// PodSet defines a flexible way to select pods in a cluster. Exactly one of the 
+// selectors should be set.  If a consumer observes none of its fields are set, 
+// they should assume an option they are not aware of has been specified and fail closed.
+type PodSet struct { 
+  // PodSelector is a labelSelector used to select Pods, This field 
+  // follows standard label selector semantics; if present but empty, it selects 
+  // all Pods.  
+  PodSelector     *metav1.LabelSelector   
+  // SameLabels is used to select a set of Pods that share the same label(s).  
+  // To be selected a Pod must have all of the labels defined in SameLabels, 
+  // If Samelabels is Empty then nothing is selected
+  SameLabels      []string
+  // NotSameLabels is used to select a set of Pods that do not have a set 
+  // of label(s). To be selected a Pod must have none of the labels defined 
+  // in NotSameLabels. If NotSameLabels is empty then nothing is selected
+  NotSameLabels   []string
 }
 
-// NamespaceMatchType describes Namespace matching strategy. 
-type NamespaceMatchType string
-
-// This list can/might get expanded in the future.
-const (
-  NamespaceMatchSelector     NamespaceMatchType = "Selector"
-  NamespaceMatchSelf         NamespaceMatchType = "Self"
-  NamespaceMatchNotSelf      NamespaceMatchType = "NotSelf"
-  NamespaceMatchSameLabels   NamespaceMatchType = "SameLabels"
-  NamespaceMatchSameLabels   NamespaceMatchType = "NotSameLabels"
-)
-
+// NamespaceSetAndPod defines a flexible way to select Namespaces and pods in a 
+// cluster.
+type NamespaceAndPodSet struct { 
+  // Namespaces is used to select a set of Namespaces.  It must be defined 
+  Namespaces   NamespaceSet 
+  // Namespaces is used to select a set of Pods in the set of Namespaces. It must 
+  // must be defined 
+  Pods         PodSet 
```

Comment:
Clarify: these 2 fields are required and must not be empty

Reply:
done